src/main/kotlin/exercise3/task3/SherlockValidatesString.kt

```diff
@@ -29,11 +31,45 @@ package exercise3.task3
  */
 
 internal fun isSherlockValid(s: String): String {
-    TODO("Implement me!!!")
+
+    val grpByChars = s.groupingBy { it }.eachCount()
+
+    val valuesOfGrouping = grpByChars.values.groupingBy { it }.eachCount()
+
+    if (valuesOfGrouping.size > 2) return "NO"
+    if (valuesOfGrouping.size == 1) return "YES"
+
+    val firstKey = valuesOfGrouping.keys.first()
+    val firstValue = valuesOfGrouping.values.first()
+    val secondKey = valuesOfGrouping.keys.last()
+    val secondValue = valuesOfGrouping.values.last()
+
+    if((secondValue == 1) and (secondKey== 1)) return "YES"
+    if((firstValue == 1) and (firstKey== 1)) return "YES"
```

Comment:
Can you elaborate on what this check is for? 
If we have the following scenario where `valuesOfGrouping` have value `{ 3 to 1, 1 to 1 }` your algorithm will return `true` even though that is a wrong value since 3 - 1 = 2.

Reply:
You can ignore the above comment; the solution is valid because we can remove that single occurrence of the string. :)